### DISCUSSION TITLE
DID 2543

### DISCUSSION THREAD

**Discussion by wangnick** [open3e/open3e#127] [Q&A] [answered]
Laut https://community.viessmann.de/t5/Waermepumpe-Hybridsysteme/Vitocal-250-A-und-Smart-Grid/td-p/347817 hat 2543 folgenden Inhalt:
- 2543.0 Raumbeheizung - Smart Grid Sollwertanhebung für Raumtemperatur
- 2543.1 Raumkühlung - Smart Grid Sollwertanhebung für Raumkühlung
- 2543.2 Trinkwassererwärmung - Smart Grid Sollwertanhebung für Trinkwassererwärmung
- 2543.3 Erwärmung Heizwasser-Pufferspeicher  - Smart Grid Sollwertanhebung für Heizbetrieb Heizwasser-Pufferspeicher
- 2543.4 Kühlung Heizwasser-Pufferspeicher - Smart Grid Sollwertabsenkung für Kühlbetrieb Heizwasser-Pufferspeicher

Zur Zeit sind im Code 2543.0 und 2543.4 unknown, und 2543.1 scheint falsch benannt zu sein.

**Comment by bu-na** [comment]
Hier mal ein Vorschlag für die Benennung der SmartGridTemperatureOffsets:

2543 : O3EComplexType(10, "SmartGridTemperatureOffsets",
         [O3EInt16(2,"SetpointIncreaseRoomTemperatureHeating"),
          O3EInt16(2, "SetpointDecreaseRoomTemperatureCooling"), 
          O3EInt16(2, "SetpointIncreaseDHWTemperature"),
          O3EInt16(2, "SetpointIncreaseBufferTemperatureHeating"),
          O3EInt16(2,"SetpointDecreaseBufferTemperatureCooling")]), #github #127 Unit K

**Reply by MyHomeMyData** [reply, marked as answer]
Ist in develop verfügbar.

**Comment by wangnick** [comment]
Sind diese Werte eventuell signed? Also, könnte man eventuell über SG die Sollwerte von Heizkreis und Warmwasser auch _absenken_? Wie kann ich das rausbekommen?

LG, Sebastian

**Comment by bu-na** [comment]
In der Serviceanleitung findet man folgendes:

**2543.0 Smart Grid Sollwertanpassung für Raumtemperatur Heizen**
Falls die potenzialfreien Kontakte für Smart Grid vom Energieversorgungsunternehmen geschaltet werden, wird der aktuelle Raumtemperatur-Sollwert um den eingestellten Wert **angehoben**.

**2543.1 Smart Grid Sollwertanpassung für Raumtemperatur Kühlen**
Falls die potenzialfreien Kontakte für Smart Grid vom Energieversorgungsunternehmen eingeschaltet werden, wird der aktuelle Raumtemperatur-Sollwert um den eingestellten Wert **abgesenkt**.

Klingt für mich eher nach unsigned. Auch aus praktischer Sicht sind eigentlich nur positive Werte sinnvoll...

Ansonsten hilft vermutlich nur ausprobieren.

Gruß Bu-Na
